src/cmd/domain.go

```diff
@@ -0,0 +1,233 @@
+	"github.com/opslevel/cli/common"
+	"github.com/opslevel/opslevel-go/v2023"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"os"
+	"strings"
+)
+
+var assignSystemCmd = &cobra.Command{
+	Use:        "system ID|ALIAS SYSTEM",
+	Short:      "Add a system to a domain",
+	Example:    `opslevel create system my_domain my_system`,
+	Args:       cobra.ExactArgs(2),
+	ArgAliases: []string{"DOMAIN_ID", "DOMAIN_ALIAS", "SYSTEM"},
+	Run: func(cmd *cobra.Command, args []string) {
+		key := args[0]
+		system := args[1]
+
+		domain, err := getClientGQL().GetDomain(key)
+		cobra.CheckErr(err)
+		common.WasFound(domain.Id == "", key)
+
+		addErr := domain.AssignSystem(getClientGQL(), system)
+		cobra.CheckErr(addErr)
+		fmt.Printf("add system '%s' to domain '%s'\n", system, domain.Name)
+	},
+}
+
+var createDomainCmd = &cobra.Command{
+	Use:   "domain",
+	Short: "Create a domain",
+	Long: `Create a domain
+
+cat << EOF | opslevel create domain -f -
+name: "My Domain"
+description: "Hello World Domain"
+ownerId: "Z2lkOi8vb3BzbGV2ZWwvVGVhbS83NjY"
+note: "Additional details"
+EOF
+`,
+	Run: func(cmd *cobra.Command, args []string) {
+		input, err := readDomainCreateInput()
+		cobra.CheckErr(err)
+		result, err := getClientGQL().CreateDomain(*input)
+		cobra.CheckErr(err)
+		fmt.Println(result.Id)
+	},
+}
+
+var deleteDomainCmd = &cobra.Command{
+	Use:        "domain ID|ALIAS",
+	Short:      "Delete a domain",
+	Long:       `Delete a domain`,
+	Args:       cobra.ExactArgs(1),
+	ArgAliases: []string{"ID", "ALIAS"},
+	Run: func(cmd *cobra.Command, args []string) {
+		key := args[0]
+		err := getClientGQL().DeleteDomain(key)
+		cobra.CheckErr(err)
+		fmt.Printf("deleted '%s' domain\n", key)
+	},
+}
+
+var getDomainCmd = &cobra.Command{
+	Use:        "domain ID|ALIAS",
+	Short:      "Get details about a domain",
+	Long:       `Get details about a domain`,
+	Args:       cobra.ExactArgs(1),
+	ArgAliases: []string{"ID", "ALIAS"},
+	Run: func(cmd *cobra.Command, args []string) {
+		key := args[0]
+		result, err := getClientGQL().GetDomain(key)
+		cobra.CheckErr(err)
+		common.WasFound(result.Id == "", key)
+		common.PrettyPrint(result)
+	},
+}
+
+var getDomainSystemCmd = &cobra.Command{
+	Use:        "system ID|ALIAS",
+	Aliases:    []string{"systems"},
+	Short:      "Get systems for a domain",
+	Long:       `The systems that belong to a domain.`,
+	Args:       cobra.ExactArgs(1),
+	ArgAliases: []string{"ID", "ALIAS"},
+	Run: func(cmd *cobra.Command, args []string) {
+		key := args[0]
+		domain, err := getClientGQL().GetDomain(key)
+		cobra.CheckErr(err)
+		common.WasFound(domain.Id == "", key)
+		resp, err := domain.ChildSystems(getClientGQL(), nil)
+		systems := resp.Nodes
+		cobra.CheckErr(err)
+		if isJsonOutput() {
+			common.PrettyPrint(systems)
+		} else {
+			w := common.NewTabWriter("Name", "ID")
+			for _, item := range systems {
+				fmt.Fprintf(w, "%s\t%s\t\n", item.Name, item.Id)
+			}
+			w.Flush()
+		}
+	},
+}
+
+var getDomainTagCmd = &cobra.Command{
```

Comment:
I don't understand the purpose of this command - tags should come through on the `get` and `list` commands data?  If they do not then we need to merge this logic.

Reply:
Same reasoning as above.